### PR TITLE
Artificer Additions - The Gearlocke pistol

### PR DESCRIPTION
**Pull request by Kyogon**
## About The Pull Request

Adds a flintlock pistol named "Gearlock" to the game, which can be made using the engineering skill. Intended to be made by Artificers. Started using bronze, and requires an extra bronze, a cog, and blacksteel bar to finish.
Adds silver arrows, bolts, and bullets, which apply the silver effect with the tradeoff of less damage and AP.
Un-capitalizes a lot of my recent additions for consistency.
Added a new bronze bar sprite.
Bronze lamptern sprite updated.

Also: Sigma and I workshopped some changes to crossbow to avoid it being a 100% crit chance. Damage factor was removed, but AP and damage on bolts were increased to just below the threshhold.

Gearlocke
A flintlock pistol that is small enough to fit on the hip. is loaded in advance similarly to a crossbow, but does less damage. Can be fired with one hand, so you can hold something in the other (like a weapon for a true pike and shot experience!)
image

## Why It's Good For The Game

Gives Artificers a unique weapon to make, which fills their role a bit more.
Identity has been changed: It will be a more portable, yet weaker version of a crossbow. Fits on a belt, but is about as strong as a bow, but slow to reload.
Silver ammo provides more utility to ranged weapons, which are woefully underused.

To make sure it is very clear: This is a groundwork update. This is not the exact state guns will be in forever. We have many plans to expand on it, but having groundwork in place gives us a bit less of a headache.

### Credits:

Oxalis for the initial Gemlock sprite
Luto for the Gearlocke version we updated it to

### CODE REVIEW

**Comment by Kavrick** [comment]
I'd rather it have a more fitting and immersive name over a joke name personally, other than that it's fine.

**Comment by bearrrrrrrr** [comment]
the gemlock

the glock, even

**Comment by kittysmooch** [comment]
i don't love the concept of a reskinned crossbow that completely negates armor but i bet you could make something cool out of existing bolt action or pipe gun code

File: code/game/objects/items/rogueweapons/ranged/ammo.dm [hunk 1]
```diff
@@ -168,28 +261,56 @@
...
 	icon = 'icons/roguetown/weapons/ammo.dmi'
 	icon_state = "musketball_proj"
-	ammo_type = /obj/item/ammo_casing/caseless/rogue/bullet
+	ammo_type = /obj/item/ammo_casing/caseless/rogue/bullet/iron
 	range = 30
 	hitsound = 'sound/combat/hits/hi_arrow2.ogg'
 	embedchance = 100
 	woundclass = BCLASS_STAB
 	flag = "bullet"
 	armor_penetration = 200
```
[the remark this answers is not in the document]
Comment (Kyogon) [reply code/game/objects/items/rogueweapons/ranged/ammo.dm:276]: This was ported over from the Matchlock, and didn't fully process how high that was. I can reduce it to be slightly higher than bolts.

**Comment by Kyogon** [comment]
> i don't love the concept of a reskinned crossbow that completely negates armor but i bet you could make something cool out of existing bolt action or pipe gun code

I'm going to tune the armor pen values. Otherwise the main benefit is that it fits on a belt hip slot. On that note though, you could easily say the same about bows and crossbows. "I don't love the concept of a reskinned bow that just has an arrow always knocked" as an example, or with melee weapons too. "I don't love the concept of a reskinned sword that has better integrity"

The list goes on. Most things in gaming are just variations of other things. That is just the nature of games.

**Comment by coffeecoffeecoffee** [comment]
I'm gonna be real with you, I severely dislike this addition and think the sprite is not up to standards - I'm sorry. The icon looks... Not great, compared to all our weapons, and the idea of essentially giving the first new gun to your new role feels not great. Maybe I'm only speaking for myself, but "gemlocks" feel way too high fantasy for what we're going for and aren't an appealing idea to me.

**Comment by bearrrrrrrr** [comment]
tbh can we get a small video or somesuch

**Comment by Disheveled-arch** [comment]
One of the few times I posted for HS PRs was for their addition of 'guns'.  I did not like them then, I do not like them now.  I think they break medieval immersion, and believe they add little to an HRP environment outside of cooky tinkers.   Great that Artificers are getting more stuff, an' I play one, yer.  But for various reasons I think this wounds the setting and brings up more lore massaging that'd need done (Are cannons in the setting yet?  Or are we still using Ballista, Catapult and Trebuchet?).  We have Crossbows, In-theme counter-parts to the gun that don't come with any of the gun baggage.  

Folk roleplay in medieval fantasy settings to get away from modern day trappings and things.  Suspend their disbelief and be somewhere fantastical and wonderous.  To deal with problems and puzzles you don't have to grapple with (usually) on a daily basis.   And in this case, Type-fuck sometimes <3.

Something about a hand-held combustion-powered, solid 'slug'/'projectile' propelling weapon is TOO modern to detach from real-world analogues and suspend disbelief for.  

SCOM: "Please, we need a healer at the inn, there's been a misfire/shooting."  <---Gross Under-exaggeration, even if it were to be an accident.

Is that a magical healers medieval-fantasy HRP wet dream, healing up gunshot wounds?  What does having guns ADD to our setting other than 'Bang' and tension against immersion?

Fuck the addition of GUNS, NOT the work that was put into the PR.  As always, many thanks to the folk who Map, Code, Sprite and Moderate.  Malum's blood run through you all. 


Wands Not Guns. 'Fetishes' Not Guns.  Blasting Rods Not Guns.  Please, anything BUT guns.

**Comment by Lutowski** [comment]
Closing this because I just think it needs more work. Our first ever gun should have that wow factor, and right now it's just a reskinned crossbow.

**Pull request closed without merging**